src/pylibsshext/scp.pyx

```diff
@@ -0,0 +1,132 @@
+            scp = libssh.ssh_scp_new(self._libssh_session, libssh.SSH_SCP_WRITE | libssh.SSH_SCP_RECURSIVE, remote_dir_b)
+            if scp is NULL:
+                raise LibsshSCPException("Allocating SCP session of remote file [%s] for write failed with error [%s]" % (remote_file, self._get_ssh_error_str()))
+
+            # Initialize the SCP channel
+            rc = libssh.ssh_scp_init(scp)
+            if rc != libssh.SSH_OK:
+                libssh.ssh_scp_free(scp)
+                raise LibsshSCPException("Initializing SCP session of remote file [%s] for write failed with error [%s]" % (remote_file, self._get_ssh_error_str()))
+
+            try:
+                # Begin to send to the file
+                rc = libssh.ssh_scp_push_file(scp, filename_b, file_size, libssh.S_IRUSR |  libssh.S_IWUSR)
+                if rc != libssh.SSH_OK:
+                    raise LibsshSCPException("Can't open remote file: %s" % self._get_ssh_error_str())
+
+                # Write to the open file
+                rc = libssh.ssh_scp_write(scp, PyBytes_AS_STRING(f.read()), file_size);
+                if rc != libssh.SSH_OK:
+                    raise LibsshSCPException("Can't write to remote file: %s" % self._get_ssh_error_str())
+            finally:
+                libssh.ssh_scp_close(scp)
+                libssh.ssh_scp_free(scp)
+
+            return libssh.SSH_OK
+
+    def get(self, remote_file, local_file):
+        cdef char *read_buffer
+
+        remote_file_b = remote_file
+        if isinstance(remote_file_b, unicode):
+            remote_file_b = remote_file.encode("utf-8")
+            remote_dir_b, filename_b = os.path.split(remote_file_b)
+
+        # Create the SCP session in read mode
+        scp = libssh.ssh_scp_new(self._libssh_session, libssh.SSH_SCP_READ, remote_file_b)
+        if scp is NULL:
+            raise LibsshSCPException("Allocating SCP session of remote file [%s] for write failed with error [%s]" % (remote_file, self._get_ssh_error_str()))
+
+        # Initialize the SCP channel
+        rc = libssh.ssh_scp_init(scp)
+        if rc != libssh.SSH_OK:
+            libssh.ssh_scp_free(scp)
+            raise LibsshSCPException("Initializing SCP session of remote file [%s] for write failed with error [%s]" % (remote_file, self._get_ssh_error_str()))
+
+        try:
+            # Request to pull the file
+            rc = libssh.ssh_scp_pull_request(scp)
+            if rc != libssh.SSH_SCP_REQUEST_NEWFILE:
+                raise LibsshSCPException("Error receiving information about file: %s" % self._get_ssh_error_str())
+
+            size = libssh.ssh_scp_request_get_size(scp)
+            mode = libssh.ssh_scp_request_get_permissions(scp)
+
+            read_buffer = <char *>PyMem_Malloc(size)
+            if read_buffer is NULL:
+                raise LibsshSCPException("Memory allocation error")
+
+            # Indicate the transfer is ready to begin
+            libssh.ssh_scp_accept_request(scp)
```

Comment:
Is return code checking required here as the function can return either `SSH_OK` or `SSH_ERROR `?